### PR TITLE
improvement: Gutenberg core blocks UI/UX design CSS compatibility

### PR DESCRIPTION
**Pull request by imnavanath**
### Description
- Added compatibility CSS for GB Column, Group, Media-Text block
- https://gist.github.com/imnavanath/76fa266f293748e8516269b7ca59f560
- Transparent header adds top 1.5em space on responsive devices

### Types of changes
- Run batch function for maintaining the design of existing users
- Added filter support as well

### How has this been tested?
- With Gutenberg plugin 
- With new WP version
- Transparent header enabling on responsive devices

### Checklist:
- [x] My code is tested
- [x] My code passes the PHPCS tests
- [x] My code follows accessibility standards
- [x] My code has proper inline documentation
- [x] I've included any necessary tests
- [x] I've included developer documentation
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: inc/class-astra-dynamic-css.php [hunk 1]
```diff
@@ -705,6 +705,125 @@ public static function return_output( $dynamic_css, $dynamic_css_filtered = '' )
...
+			if ( self::gutenberg_core_blocks_css_comp() ) {
+
+				/**
+				 * If transparent header is activated then it adds top 1.5em padding space, so this CSS will fix this issue.
+				 * This issue is only visible on responsive devices.
+				 *
+				 * @since x.x.x
+				 */
+				$trans_header_responsive_top_space_css_fix = array(
+					'.ast-theme-transparent-header #primary, .ast-theme-transparent-header #secondary' => array(
```
Comment (premanshup) [inline inc/class-astra-dynamic-css.php:717]: @imnavanath Why haven't we used these similar conditions here? - https://github.com/brainstormforce/astra/commit/eedd4f32f045639485f30d845c457c5e175df6dc

File: inc/class-astra-dynamic-css.php [hunk 2]
```diff
@@ -705,6 +705,125 @@ public static function return_output( $dynamic_css, $dynamic_css_filtered = '' )
...
+
+				/* Parse CSS from array() -> max-width: (tablet-breakpoint)px CSS */
+				$parse_css .= astra_parse_css( $trans_header_responsive_top_space_css_fix, '', astra_get_tablet_breakpoint() );
+
+				$desktop_screen_gb_css = array(
+					// Group block, Columns block, Gallery block, Table block & has-text-align-center selector compatibility Desktop CSS.
+					'.wp-block-columns'                  => array(
+						'margin-bottom' => 'unset',
+					),
+					'figure.size-full'                   => array(
```
Comment (premanshup) [inline inc/class-astra-dynamic-css.php:730]: @imnavanath Can we add any prefix class here? Maybe related to `wp-block`. figure.size-full seems like very common CSS class.

**Review by premanshup — CHANGES_REQUESTED** [review]

File: inc/class-astra-dynamic-css.php (hunk 1 above)
Comment (imnavanath) [reply inc/class-astra-dynamic-css.php:717]: Correct 👍 Doing it WIP...

File: inc/class-astra-dynamic-css.php (hunk 2 above)
Comment (imnavanath) [reply inc/class-astra-dynamic-css.php]: Unfortunately, the image directly applies `figure` block, no wrapper here used by GB (like `wp-block` etc). Here we can improve CSS from `figure.size-full` to `.wp-block-image.size-full`. I've done this.

File: inc/class-astra-dynamic-css.php (hunk 1 above)
Comment (imnavanath) [reply inc/class-astra-dynamic-css.php]: Done

**Review by premanshup — APPROVED** [review]

**Review by shubhw12 — APPROVED** [review]

**Pull request merged**